lib/src/index.rs

```diff
@@ -75,6 +75,14 @@ pub trait ReadonlyIndex: Send + Sync {
 
     fn as_index(&self) -> &dyn Index;
 
+    // TODO: might be better to split Index::evaluate_revset() to
+    // Readonly/MutableIndex::evaluate_static().
+    fn evaluate_revset_static(
+        &self,
+        expression: &ResolvedExpression,
+        store: &Arc<Store>,
+    ) -> Result<Box<dyn Revset<'static>>, RevsetEvaluationError>;
+
```

Comment:
> I'm not sure if this is feasible for the other backends, but I guess there
would be less lifetimed variables than DefaultReadonlyIndex.

Yes, the index we have at Google doesn't have a lifetime variable (since it's server-backed)